webpack/components/AnsibleRoles/components/AnsibleRolesImportButtonWrapper.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { useAPI } from 'foremanReact/common/hooks/API/APIHooks';
+import { showToast } from '../../../toastHelper';
+import { AnsibleRolesImportButton } from './AnsibleRolesImportButton';
+
+export const AnsibleRolesImportButtonWrapper = () => {
+  let isButtonDisabled = true;
+  const {
+    response: { results },
+    status,
+  } = useAPI('get', '/api/v2/permissions/current_permissions');
+
+  if (status === 'ERROR') {
+    showToast({
+      type: 'error',
+      message: __('There was an error requesting user permissions'),
+    });
+    return null;
+  } else if (status === 'RESOLVED') {
+    if (
+      results.some(permission => permission.name === 'import_ansible_roles')
+    ) {
+      isButtonDisabled = false;
+    }
+  }
+  return <AnsibleRolesImportButton isDisabled={isButtonDisabled} />;
```

Comment:
We usually hide buttons if the user has no permission, so returning null instead of a disabled button. Can you send the permission as part of the ui_ansible_roles api, to avoid extra api call

Reply:
The button is now hidden instead of just disabled.
I'm afraid, integrating the request for the permission into the request to fetch roles is not easily possible as it is made by TableIndexPage.